src/content/en/updates/2018/07/page-lifecycle-api.md

```diff
@@ -0,0 +1,962 @@
+](#developer-recommendations-for-each-state), we've released
+[PageLifecycle.js](https://github.com/GoogleChromeLabs/page-lifecycle), a
+JavaScript library for observing Page Lifecycle API state changes.
+
+[PageLifecycle.js](https://github.com/GoogleChromeLabs/page-lifecycle)
+[normalizes cross-browser differences in event firing order so that state change
+[always occur exactly as outlined in the chart and tables in this article (and
+[do so consistently in all browsers).
+
+
+## Developer recommendations for each state {: #developer-recommendations-for-each-state}
+
+As developers, it's important to both understand Page Lifecycle states &mdash;
+as well as know how to observe them in code &mdash; because the type of work you
+should (and should not) be doing depends largely on what state your page is in.
+
+For example, it clearly doesn't make sense to be updating the inline styles of
+DOM elements if your page is in the hidden state. While this example is pretty
+obvious, there are other recommendations that aren't so obvious that are worth
+enumerating.
+
+<table>
+  <tr>
+    <th>State</th>
+    <th>Developer recommendations</th>
+  </tr>
+  <tr>
+    <td><strong>Active</strong></td>
+    <td>
+      <p>The active state is the most critical time for the user and thus the
+      most important time for your page to be
+      <a href="/web/updates/2018/05/first-input-delay">
+      responsive to user input</a>.</p>
+      <p>Any non-UI work that may block the main thread should be deprioritized
+      to <a href="/web/updates/2015/08/using-requestidlecallback">
+      idle periods</a> or <a
+      href="/web/fundamentals/performance/rendering/optimize-javascript-execution#reduce_complexity_or_use_web_workers">
+      offloaded to a web worker</a>.</p>
+    </td>
+  </tr>
+  <tr>
+    <td><strong>Passive</strong></td>
+    <td>
+      <p>In the passive state the user is not interacting with the page, but
+      they can still see it. This means UI updates and animations should still
+      be smooth, but the timing of when these updates occur is less critical.</p>
+      <p>When the page changes from <em>active</em> to <em>passive</em>, it's a
+      good time to persist unsaved application state.</p>
+    </td>
+  </tr>
+  <tr>
+    <td><strong>Hidden</strong></td>
+    <td>
+      <p>When the page changes from <em>passive</em> to <em>hidden</em>, it's
+      possible the user will not interact with it again until it's reloaded.</p>
+      <p>The transition to <em>hidden</em> is also often the last state change
+      that's reliably observable by developers (this is especially true on
+      mobile, as users can close tabs or the browser app itself, and the
+      pagehide and unload events are not fired in those cases).</p>
+      <p>This means you should treat the hidden state as the likely end to the
+      user's session &mdash; including persisting any unsaved application state,
```

Comment:
'including' cannot modify the verb 'treat' in this sentences. Start a new sentence with 'In other words, persist any ...'

Reply:
Done.